content/pixel_fifo.md

```diff
@@ -0,0 +1,252 @@
+# Pixel FIFO
+Before we get started, all references to a cycle are meant as T-cycles
+(4.19 MHz) and cycle counts are doubled on CGB in double speed mode. Also
+when stating that a certain action "lengthens mode 3" this means that
+mode 0 (hblank) is shortened. The extra time has to be made up somewhere
+and the lengthening of mode 3 eats into the mode that comes afterwards,
+which happens to be mode 0 as shown in the following diagram.
+
+![](imgs/game-boy-lcd-refresh-diagram.png "imgs/game-boy-lcd-refresh-diagram.png")
+
+## FIFO info
+FIFO stands for First In, First Out. The first pixel to be pushed to the
+FIFO is the first pixel to be popped off. In theory that sounds great,
+in practice there are a lot of intricacies.
+
+There are two pixel FIFOs. One for background pixels and one for OAM
+(sprite) pixels. These two FIFOs are not shared. They are independent
+of each other. The two FIFOs are mixed only when popping items. Sprites
+take priority unless they're transparent (color 0) which will be
+explained in detail later. The FIFOs can each hold up to 16 pixels at a
```

Comment:
"The FIFOs can each hold up to 16 pixels at a time" This doesn't seem relevant here. As this is quite fundamental I'd recommend having a paragraph before this explaining it. e.g "Each FIFO can hold up to 16 pixels. The FIFO and Pixel Fetcher work together to ensure that the FIFO always contains at least 8 pixels at any given time, as 8 pixels are required for the Pixel Rendering operation to take place."